src/confluent_kafka/aio/_AIOProducer.py

```diff
@@ -17,40 +17,122 @@
-        self._running = False
-        if auto_poll:
-            self._running = True
-            self._running_loop = asyncio.create_task(self._loop())
+        
+        # Metrics for latency tracking
+        self._latency_metrics = []
+        self._total_calls = 0
+        self._total_latency = 0.0
+        self._min_latency = float('inf')
+        self._max_latency = 0.0
+        
+        # Batching configuration and buffer
+        self._batch_size = batch_size
+        self._message_buffer = []
+        self._buffer_futures = []  # Track futures for each buffered message
+        self._buffer_lock = asyncio.Lock()  # Protect buffer access
+    
 
-    async def stop(self):
-        if self._running:
-            self._running = False
-            await self._running_loop
+    # ========================================================================
+    # BLOCKING OPERATIONS - Use ThreadPool to avoid blocking event loop
+    # These operations may block waiting for network I/O or other resources
+    # ========================================================================
 
-    async def _loop(self):
-        while self._running:
-            await self.poll(1.0)
+    async def poll(self, timeout=0, *args, **kwargs):
+        """Processes delivery callbacks from librdkafka - blocking behavior depends on timeout
+        
+        This method triggers any pending delivery callbacks (on_delivery) that have been
+        queued by librdkafka when messages are delivered or fail to deliver.
+        
+        Args:
+            timeout: Timeout in seconds for waiting for callbacks:
+                    - 0 = non-blocking, return immediately after processing available callbacks
+                    - >0 = block up to timeout seconds waiting for new callbacks to arrive
+                    - -1 = block indefinitely until callbacks are available
+        
+        Returns:
+            Number of callbacks processed during this call
+        """
+        if timeout > 0 or timeout == -1:
+            # Blocking call - use ThreadPool to avoid blocking event loop
+            return await self._call(self._producer.poll, timeout, *args, **kwargs)
 
-    async def poll(self, *args, **kwargs):
-        await self._call(self._producer.poll, *args, **kwargs)
 
+    async def produce(self, topic, value=None, key=None, *args, **kwargs):
+        """Batched produce: Accumulates messages in buffer and flushes when threshold reached
+        
+        Args:
+            topic: Kafka topic name (required)
+            value: Message payload (optional)
+            key: Message key (optional)
+            *args, **kwargs: Additional parameters like partition, timestamp, headers
+        """
+        # Start Time
+        start_time = time.perf_counter()
+        
+        result = asyncio.get_running_loop().create_future()
+        user_callback = kwargs.get('on_delivery')
+        
+        await self._buffer_lock.acquire()
```

Comment:
This is not necessary. Only one thread will be accessing this area at once as an async function. Even if multiple threads are using producers they would have separate internal state.

Reply:
The time you'd need an asyncio.Lock would be if you awaited in the middle of working on memory objects that might be manipulated by the other async boundary which we don't need as we're done with manipulation by the time we might do `_flush_buffer`. This lock isn't threadsafe btw for preserving any threaded operations if that was the intent (which I don't think we should guard against as per the prior comment)